packages/core/src/interpreter.ts

```diff
@@ -186,12 +186,17 @@ export class Actor<TLogic extends AnyActorLogic>
       type: '@xstate.actor',
       actorRef: this
     });
-    this._initState(options?.state);
 
-    if (systemId && (this._state as any).status === 'active') {
+    if (systemId) {
       this._systemId = systemId;
       this.system._set(systemId, this);
     }
+
+    this._initState(options?.state);
+
+    if (systemId && (this._state as any).status !== 'active') {
```

Comment:
Nit: do we still need `any`?

Reply:
Yes. The whole `SnapshotFrom<TLogic>` pattern (and that's how `this._state` is typed here) just makes it impossible to use *anything* on it. This type is a deferred conditional here since `TLogic` is generic and TS can't guarantee that it won't resolve to `never`... and we can't access anything on `never`. So it errors on any property access here since that's a situation that can happen from its PoV. I think that we shouldn't use this pattern in our signatures/classes.